### PR TITLE
Improve error handling and pipe flow for `just ps`

### LINKED ISSUE #1899: `ps` recipe does not handle errors causing dependent recipes to needlessly fail
## Description

The `ps` recipe sometimes fails and when it does, it causes the dependent recipes to fail, even though `ps` isn't truly critical. It is nice but it failing isn't a reason to prevent downstream parts of the recipe that called it to not run.

## Reproduction

I'm not sure exactly how to reproduce it consistently, but here is example output from when it failed for me locally, causing the tests to never run:

```
➜  openverse git:(add/ignore-non-main-target-prs) ✗ just catalog/test -k pr_review_reminders
just _mount-test "bash -c \'pytest -k pr_review_reminders\'"
env COMPOSE_PROFILES="catalog_dependencies" just ../up 
env COMPOSE_PROFILES="catalog_dependencies" docker-compose -f docker-compose.yml up -d
[+] Running 3/3
 ✔ Container openverse-s3-1           Started                                                      0.6s 
 ✔ Container openverse-upstream_db-1  Started                                                      0.6s 
 ✔ Container openverse-load_to_s3-1   Started                                                      0.8s 

python3 utilities/ps.py
================================================================================
                                 Service Ports                                  
================================================================================
Traceback (most recent call last):
  File "/home/sara/projects/openverse/utilities/ps.py", line 105, in <module>
    print_ps()
  File "/home/sara/projects/openverse/utilities/ps.py", line 99, in print_ps
    for service in parse_ps():
  File "/home/sara/projects/openverse/utilities/ps.py", line 66, in parse_ps
    service, image, ports = line.split("\t")
ValueError: not enough values to unpack (expected 3, got 2)
error: Recipe `ps` failed on line 153 with exit code 1
error: Recipe `up-deps` failed on line 64 with exit code 1
error: Recipe `test` failed on line 111 with exit code 1
```

A fix for this might be as simple as changing the `ps` recipe to ignore the exit code of the Python script:

```
# List all services and their URLs and ports
ps:
    python3 utilities/ps.py &
``` 

A better fix would include that but also add cleaner error handling to the Python script itself.

### PR DESCRIPTION
**Pull request by AetherUnbound**
## Fixes

Fixes #1899 by @sarayourfriend

## Description

This PR makes two changes to the `just ps` recipe:

1. It changes the recipe itself so that it never reports a failing exit code, meaning it doesn't fail any subsequent jobs
2. It adds some additional error handling to the `ps.py` script so that errors are more clearly reported. 

Here's an example of a failing run:

```
$ j catalog/up "s3 upstream_db"
env COMPOSE_PROFILES="catalog" just ../up s3 upstream_db
env COMPOSE_PROFILES="catalog" docker-compose -f docker-compose.yml -f docker-compose.overrides.yml up -d s3 upstream_db
[+] Running 3/3
 ⠿ Network openverse_default          Created                                                                                                                                                                  0.1s
 ⠿ Container openverse-upstream_db-1  Started                                                                                                                                                                  0.5s
 ⠿ Container openverse-s3-1           Started                                                                                                                                                                  0.6s

python3 utilities/ps.py || true
================================================================================
                                 Service Ports                                  
================================================================================
Traceback (most recent call last):
  File "/home/madison/git-a8c/openverse/utilities/ps.py", line 104, in print_ps
    for service in parse_ps():
  File "/home/madison/git-a8c/openverse/utilities/ps.py", line 69, in parse_ps
    raise ValueError(f"Unexpected line format from `docker ps`: {line}")
ValueError: Unexpected line format from `docker ps`: openverse-upstream_db-1	openverse-upstream_db	5432/tcp

================================================================================
Failed to get service info (ValueError): 
Unexpected line format from `docker ps`: openverse-upstream_db-1	openverse-upstream_db	5432/tcp
================================================================================
```

## Testing Instructions

Make the following change to `ps.py`:

```diff
diff --git a/utilities/ps.py b/utilities/ps.py
index cdf1d7a66..30ca50f1c 100644
--- a/utilities/ps.py
+++ b/utilities/ps.py
@@ -64,7 +64,7 @@ def parse_ps() -> list[Service]:
     lines = get_ps()
     lines = [stripped for line in lines if (stripped := line.strip())]
     for line in lines:
-        parts = line.split("\t")
+        parts = line.split("\t")[:1]
         if len(parts) != 3:
             raise ValueError(f"Unexpected line format from `docker ps`: {line}")
         service, image, ports = parts
```

Then run `just up` and observe an error similar to the above.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by obulat — APPROVED** [review]
Thank you for fixing this!
I don't quite understand what the problem with the example output (`ValueError: Unexpected line format from `docker ps`: openverse-upstream_db-1	openverse-upstream_db	5432/tcp`) is. It would be nice if we could make the parsing more flexible to fix it, too, so that we wouldn't need to handle errors.

File: justfile [hunk 1]
```diff
@@ -150,7 +150,7 @@ build *args:
 
 # List all services and their URLs and ports
 ps:
-    python3 utilities/ps.py
+    python3 utilities/ps.py || true
```
Comment (dhruvkb) [inline justfile:153]: A comment on why we're silently exiting with code 0 would be nice. I don't know if there are any recipes that depend on `ps`.

**Review by dhruvkb — APPROVED** [review]

File: justfile (hunk 1 above)
Comment (AetherUnbound) [reply justfile:153]: Great call, I'll add that!

**Comment by AetherUnbound** [comment]
>I don't quite understand what the problem with the example output (`ValueError: Unexpected line format from docker ps: openverse-upstream_db-1 openverse-upstream_db 5432/tcp`) is. It would be nice if we could make the parsing more flexible to fix it, too, so that we wouldn't need to handle errors.

This was just my example from forcing the error, it's not an organic issue itself! I think the issue Sara was seeing was that the `ps` was happening so quickly after `up` that there weren't actually any services to report, and so `ps` wasn't able to parse the output properly. You're right, that example line _is_ correct and we do parse that correctly. I'm not sure what the exact output Sara was encountering which cause the initial reported error, so I wouldn't be certain how best to adjust the parsing to compensate.

**Comment by AetherUnbound** [comment]
This is failing because of #2055, I think it's safe to merge nonetheless

**Pull request merged**